### PR TITLE
use retries for flakey tests

### PR DESCRIPTION
**Pull request by graingert**
@westy92 currently the tests are a bit flakey (running the tests on master often fail)

### CODE REVIEW

**Comment by graingert** [comment]
@westy92 ok this one actually passes every time, for real this time.

File: test/index.ts [hunk 1]
```diff
@@ -52,44 +52,46 @@ describe('HtmlPdf', () => {
...
       let launchStub: sinon.SinonStub;
       const error = new Error('failed!');
-      try {
-        launchStub = sinon.stub(chromeLauncher, 'launch').callsFake(() => Promise.reject(error));
-        await HtmlPdf.create('<p>hello!</p>');
-        expect.fail();
-      } catch (err) {
-        expect(err).to.equal(error);
-      } finally {
+      beforeEach(() => {
```
Comment (westy92) [inline test/index.ts]: We should only need to setup the stub once.  Change this `beforeEach` and `afterEach` to just `before` and `after`.
We'll need to also add:
```typescript
afterEach(() => {
  // reset stub call counters
  launchStub.reset();
});
```

File: test/index.ts [hunk 2]
```diff
@@ -52,44 +52,46 @@ describe('HtmlPdf', () => {
...
+        launchStub = sinon
+          .stub(chromeLauncher, 'launch')
+          .callsFake(() => Promise.reject(error));
+      });
+
+      afterEach(() => {
         launchStub.restore();
-      }
-    });
+        launchStub = undefined;
```
Comment (westy92) [inline test/index.ts]: I don't think we need this line.

File: test/index.ts [hunk 3]
```diff
@@ -52,44 +52,46 @@ describe('HtmlPdf', () => {
...
+      });
 
-    it('should use running Chrome to generate a PDF (specify port)', async () => {
-      const launchStub = sinon.stub(chromeLauncher, 'launch');
-      try {
+      it('should handle a Chrome launch failure', async () => {
```
Comment (westy92) [inline test/index.ts]: Let's keep this test more inline with what it was before:
```typescript
    it('should handle a Chrome launch failure', async () => {
      try {
        await HtmlPdf.create('<p>hello!</p>');
        expect.fail();
      } catch (err) {
        expect(err).to.equal(error);
      }
    });
```

File: test/index.ts (hunk 1 above)
Comment (westy92) [reply test/index.ts]: Setting up the stub, if we do it once, may not even require a `before` block.

**Comment by westy92** [comment]
Tests have been stabilized with the v0.4.0 release.

**Pull request closed without merging**